kafka-streams-framework/src/main/java/org/hypertrace/core/kafkastreams/framework/KafkaStreamsApp.java

```diff
@@ -302,4 +307,20 @@ private Duration getShutdownDuration() {
         ? config.getDuration(SHUTDOWN_DURATION)
         : Duration.ofSeconds(30);
   }
+
+  private void delayStartup(long initialDelayMillis) {
+    if (initialDelayMillis > 0) {
+      isSleeping = true;
+      try {
+        getLogger()
+            .info(
+                "Sleeping for {} millisecond before kafka streams app is started.",
+                initialDelay.toMillis());
```

Comment:
once millis is calculated, no need to recalculate

Reply:
Done